#107 Create Billing Period Start dimension (#120)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/dimensions/billingperiodstart.md

# Billing Period Start

> Billing period represents the time window for which you have or will receive an invoice for. Billing period represents the time window for which you have or will receive an invoice for. The time window is inclusive of the start date and exclusive of the end date.

Reviewer: Change to:
Billing period represents the time window for which an organization have or will receive an invoice for. The time window is inclusive of the start date and exclusive of the end date.

> Billing period represents the time window for which you have or will receive an invoice for. Billing period represents the time window for which you have or will receive an invoice for. The time window is inclusive of the start date and exclusive of the end date.

Reviewer: ```suggestion
Billing period represents the time window for which an organization have or will receive an invoice for. The time window is inclusive of the start date and exclusive of the end date.
```

Reviewer 2: Given we're using US English spelling, we should probably use US English grammar and change "organization have" to "organization has".

> Billing Period Start represents the beginning of a billing period.

Reviewer 2: Should we be explicit about it being a date and time? I think we should use "the" since this applies to the value for the row.
```suggestion
Billing Period Start represents the start date and time of the billing period.
```

> The BillingPeriodStart column MUST be present in the billing data. This column MUST conform to [FOCUS Date/Time Requirements](#date/timerequirements) and MUST NOT contain null values. The sum of the Billed Cost metric for line items in a given billing period MUST match the total cost of the invoices received for that billing period.

Reviewer 3: ```suggestion
The BillingPeriodStart column MUST be present in the billing data. This column MUST conform to [FOCUS Date/Time Requirements](#datetimerequirements) and MUST NOT contain null values. The sum of the Billed Cost metric for line items in a given billing period MUST match the total cost of the invoices received for that billing period.
```

> The BillingPeriodStart column MUST be present in the billing data. This column MUST conform to [FOCUS Date/Time Requirements](#date/timerequirements) and MUST NOT contain null values. The sum of the Billed Cost metric for line items in a given billing period MUST match the total cost of the invoices received for that billing period.

Reviewer: Is the casing for "Billed Cost metric" correct? Should it be "billed cost metric"

Reviewer 3: This is correct. If we're referring to a concept (e.g. billing account, resource), we can avoid capitalizing it. When we're referring to the actual column we use the display name.

## Column ID

> BillingPeriodStart

Reviewer 2: Is anyone else have a nagging thought every time you see this that it should be "BillingPeriodStart**Date**"? For some reason, I can't look at this without feeling like it's a typo.

Author: I did think about this, but my take is that BillingPeriodStart works fine. If we say Date, it might mean it is a column of type Date and not Datetime. Less confusing this way.

Reviewer 3: Can. Shorter name would be nice. It feels like Date would imply its not a DateTime per the expectation?

## Display Name

Billing Period Start

## Description

The beginning of a billing period.

## Content Constraints

| Constraint      | Value                                                                |
|:----------------|:---------------------------------------------------------------------|
| Column Required | True                                                                 |
| Data type       | Datetime                                                             |
| Allows nulls    | False                                                                |

> | Value format    | Meets [FOCUS Date/Time Format] (#date/timerequirements) requirements |

Reviewer 3: ```suggestion
| Value format    | Meets [FOCUS Date/Time Format] (#datetimerequirements) requirements |
```
